testing/cli/test_sct_run_batch.py

```diff
@@ -12,46 +13,43 @@
 from spinalcordtoolbox.scripts import sct_run_batch
 
 
-def write_dummy_script(script):
-    """
-    Dummy script that displays subject
-    :param script: NamedTemporaryFile object. Script to write
-    :return:
-    """
+@pytest.fixture
+def dummy_script(tmp_path):
+    """Dummy executable script that displays subject."""
+    path_out = str(tmp_path / "dummy_script.sh")
     script_text = """
     #!/bin/bash
     SUBJECT=$1
     echo $SUBJECT
     """
-    # indexing removes beginning newline
-    script.write(dedent(script_text)[1:])
-    script.flush()
+    with open(path_out, 'w') as script:
+        # indexing removes beginning newline
+        script.write(dedent(script_text)[1:])
+        script.flush()
+    return path_out
 
 
-def test_config_with_args_warning():
-    # This is formatted strangely because pep8 says so.
-    with \
-        NamedTemporaryFile('w', suffix='.json') as config,\
-            TemporaryDirectory() as data,\
-            TemporaryDirectory() as out:
+def test_config_with_args_warning(tmp_path):
+    data = str(tmp_path / 'data')
+    out = str(tmp_path / 'out')
+    config_path = str(tmp_path / 'config.json')
 
-        cnf_text = '{{"jobs": 1, "path_data": "{}", "path_output": "{}"}}'.format(data, out)
-        config.write(cnf_text)
-        config.flush()
+    with open(config_path, 'w') as config:
+        cnf_text = {"jobs": 1, "path_data": data, "path_output": out}
+        json.dump(cnf_text, config)
 
-        with pytest.warns(UserWarning, match=r'-config.*discouraged'):
-            # I'm not sure how to check that argparse is printing the right error here, but I trust
-            with pytest.raises(FileNotFoundError):
-                sct_run_batch.main(['-c', config.name, '-include', 'something', '-script', 'script'])
+    with pytest.warns(UserWarning, match=r'-config.*discouraged'):
+        # I'm not sure how to check that argparse is printing the right error here, but I trust
+        with pytest.raises(FileNotFoundError):
+            sct_run_batch.main(['-c', config_path, '-include', 'something', '-script', 'script'])
```

Comment:
Why not use your new `dummy_script` fixture? From the name of this test, it's more about the `UserWarning` than about the `FileNotFoundError`. (I'm assuming the `FileNotFoundError` is about `-script script`, rather than, say, non-existent data.)

Reply:
Good call! 

In fact, this even revealed another bug in the test, since now this test actually _runs_ `sct_run_batch` using the `path_data` and `path_output` inputs. (And, I forgot to `mkdir` the `data` and `out` directories, aha.)